src/cfg.rs

```diff
@@ -28,6 +28,7 @@ cfg_proto! {
         ($($item:item)*) => {
             cfg_feature! {
                 #![feature = "client"]
+                #[cfg(not(target_os = "wasi"))] // seems not to work
```

Comment:
For some reason, the compiler still attempts to build the code behind the `cfg_client!` macro despite including the !Wasi config line.